src/coreclr/jit/dataflow.h

```diff
@@ -58,19 +58,47 @@ void DataFlow::ForwardAnalysis(TCallback& callback)
         }
         else
         {
-            FlowEdge* preds = m_pCompiler->BlockPredsWithEH(block);
-            for (FlowEdge* pred = preds; pred; pred = pred->getNextPredEdge())
+            for (FlowEdge* pred = block->bbPreds; pred; pred = pred->getNextPredEdge())
             {
                 callback.Merge(block, pred->getSourceBlock(), pred->getDupCount());
             }
         }
 
         if (callback.EndMerge(block))
         {
-            block->VisitAllSuccs(m_pCompiler, [&worklist](BasicBlock* succ) {
+            // The clients using DataFlow (CSE, assertion prop) currently do
+            // not need EH successors here:
+            //
+            // 1. CSE does not CSE into handlers, so it considers no
+            // expressions available at the beginning of handlers;
+            //
+            // 2. Assertion prop never kills any assertions, so it is
+            // sufficient to propagate facts from the 'try' head block, since
+            // that block dominates all other blocks in the 'try'. That happens
+            // implicitly below.
+            //
+            block->VisitRegularSuccs(m_pCompiler, [&worklist](BasicBlock* succ) {
                 worklist.insert(worklist.end(), succ);
                 return BasicBlockVisit::Continue;
             });
         }
+
+        if (m_pCompiler->bbIsTryBeg(block))
```

Comment:
Why do we not need to consider the exceptional successors of filters here?

(Not saying it should)

Reply:
Do you mean the second pass EH successors?

Reply:
Yes.

Reply:
I think that because the filter and its enclosed handlers are all dominated by a same try entry there is no need to consider the filter as a pred of those handlers.